### ISSUE TITLE
Error: Chunk.entry was removed. Use hasRuntime()

### ISSUE THREAD

**Issue by jameswyse** [BohdanTkachenko/webpack-split-by-path#20]
The latest webpack release (2.1.0-beta.17) raises the following error:

```
node_modules/webpack/lib/Chunk.js:34
        throw new Error("Chunk.entry was removed. Use hasRuntime()");
        ^

Error: Chunk.entry was removed. Use hasRuntime()
    at Chunk.Object.defineProperty.get (node_modules/webpack/lib/Chunk.js:34:9)
    at node_modules/webpack-split-by-path/index.js:91:23
    at Array.filter (native)
    at Compilation.<anonymous> (node_modules/webpack-split-by-path/index.js:89:10)
    at Compilation.applyPluginsBailResult (node_modules/webpack/node_modules/tapable/lib/Tapable.js:52:27)
    at Compilation.seal (node_modules/webpack/lib/Compilation.js:543:8)
    at Compiler.<anonymous> (node_modules/webpack/lib/Compiler.js:436:15)
    at node_modules/webpack/node_modules/tapable/lib/Tapable.js:152:11
    at Compilation.<anonymous> (node_modules/webpack/lib/Compilation.js:434:10)
    at node_modules/webpack/lib/Compilation.js:410:12
    at node_modules/webpack/lib/Compilation.js:325:10
    at node_modules/async/lib/async.js:52:16
    at done (node_modules/async/lib/async.js:246:17)
    at node_modules/async/lib/async.js:44:16
    at node_modules/webpack/lib/Compilation.js:325:10
    at node_modules/async/lib/async.js:52:16
```

Seems to be related to [index.js:91](https://github.com/BohdanTkachenko/webpack-split-by-path/blob/master/index.js#L91)

Also see this related issue on the webpack repo: https://github.com/webpack/webpack/issues/2764

And this commit to the extract-text plugin for a similar problem may be of help: https://github.com/webpack/extract-text-webpack-plugin/commit/ba0a71dcc72dd298f602f96906e2042a08cbe039

**Comment by meirzadotcom** [comment]
@jameswyse try this:

```
npm install extract-text-webpack-plugin@^2.0.0-beta
```

And if you see new error (below) after installation:

```
throw new Error("Breaking change: extract now only takes a single argument. 
              ^
Error: Breaking change: extract now only takes a single argument. Either an options object *or* the loader(s).
```

Try to change your module loaders in webpack config:

```
loader: ExtractTextPlugin.extract({ notExtractLoader: 'style-loader', loader: 'css-loader!postcss-loader!sass-loader' })
```

**Comment by salzhrani** [comment]
The Chunk API changed. it seems that it needs a fix

**Comment by ndelangen** [comment]
I'm getting the same Error @jameswyse is getting with the newest version of webpack

```
"webpack": "^2.1.0-beta.20",
"webpack-dev-server": "^2.1.0-beta.0",
"extract-text-webpack-plugin": "^2.0.0-beta.3",
```

**Comment by ndelangen** [comment]
I think this is what's failing:
https://github.com/BohdanTkachenko/webpack-split-by-path/blob/675c9a49b3a11357b0e21c47ef71f6e14e7fe775/index.js#L91

According to this thread:
https://github.com/webpack/webpack/issues/2764

This fix in `extract-text-webpack-plugin` fixed something related to it:
https://github.com/webpack/extract-text-webpack-plugin/commit/d40c13115f1d591558cbe4837b570642356cc178

**Comment by ndelangen** [comment]
> fixed by changing if(c.entry) to if(c.hasRuntime())

found here: https://github.com/Hashnode/mern-starter/issues/200#issuecomment-234610731

**Comment by ndelangen** [comment]
I was able to find a solution to make it work @jameswyse, I've submitted a PR, If in need, you could see what I did to make it work.

**Comment by aldendaniels** [comment]
I'm running into this issue using webpack `v1.13.2`

**Comment by tarlepp** [comment]
yep same here with webpack 1.13.2

**Comment by tarlepp** [comment]
@aldendaniels you can get this working if you change your package.json to contain following:

```
"webpack-split-by-path": "0.0.10"
```

It seems like #22 broke compatibility with webpack 1.x branch. That PR should also contain updated package.json but it doesn't.

**Comment by BohdanTkachenko** [comment]
You're right, @tarlepp Last version is 0.1.0-beta.1 which is working only with webpack2
But that's a bit stange to me, because NPM should not update to major version automatically

**Comment by tarlepp** [comment]
@BohdanTkachenko actually it's _minor_ change http://semver.org/

**Comment by BohdanTkachenko** [comment]
@tarlepp you're right. I will fix this then

**Comment by BohdanTkachenko** [comment]
We just released 2.0.0 version of this plugin for webpack 2. 
For webpack 1 there is still a branch webpack1 and version 1.0.0 on NPM.

Please consider upgrading to webpack 2 and version 2.0.0 of this plugin.
If this issue still reproduces on 2.0.0 please re-open this issue.
If you really need to have this in 1.0.0, PR's to webpack1 branch are always welcome, but I don't longer plan to fix issues for webpack1.
